tiledb/api/src/array/attribute/strategy.rs

```diff
@@ -117,7 +117,7 @@ pub fn prop_attribute(
         .map(|d| Just(d).boxed())
         .unwrap_or(prop_datatype_implemented().boxed());
 
-    datatype.prop_flat_map(move |datatype| {
+    datatype.prop_ind_flat_map(move |datatype| {
```

Comment:
Can you write a few notes on why this change is required? Reading the docs all I'm seeing is that it creates an independent strategy that then can't be shrunk.

Reply:
As the tests were failing it was seeming like the schema wasn't really shrinking (the focus test of this PR would fail and print "minimum input is <thing>" with something that was very obviously not the minimum input).  So I was playing with shrinking for a bit and came to this as a way to simplify it a bit - we don't want to bother shrinking the datatype since the datatype is fixed.

(it turns out that it just doesn't shrink quickly enough and the default is only 1024 iterations, I did observe smaller schemata when I increased the iterations, but it doesn't change very much at each iteration so it goes very slowly. If we implement our own shrinking we probably ought to be more aggressive with shrinking cutting a lot, and complicating growing back a little)

I think at the time that I wrote this I misunderstood a bit what shrinking the datatype would actually mean.  But I think what happens under the hood In this particular case is as we try to simplify the attribute, one of the steps might find its way here and try to simplify the datatype; with `prop_ind_flat_map` it stops there, whereas with `prop_flat_map` we have a different ValueTree type. If we have `Just(dt)` then that doesn't do anything, if we have `prop_datatype_implemented()` which is basically "choose one of these things at random" then we also don't do anything.

So we save an indirect function call per simplification step, which is pretty much epsilon given that it also wants to re-run the whole test at each step.

Looks like all the `prop_ind_flat_map`s are ranging on a datatype strategy so in the end this doesn't really do anything meaningful.